solidity-checks/src/main/java/org/sonarsource/solidity/checks/CognitiveComplexityVisitor.java

```diff
@@ -17,24 +17,25 @@
 import org.sonarsource.solidity.frontend.SolidityParser.SourceUnitContext;
 import org.sonarsource.solidity.frontend.SolidityParser.StatementContext;
 import org.sonarsource.solidity.frontend.SolidityParser.WhileStatementContext;
-import org.sonarsource.solidity.frontend.Utils;
 
 public class CognitiveComplexityVisitor extends SolidityBaseVisitor<Token> {
   private int complexity;
   private int nestingLevel;
-  protected Map<String, Integer> functionsComplexity;
+  public Map<FunctionDefinitionContext, Integer> functionsComplexity = new HashMap<>();;
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Make functionsComplexity a static final constant or non-public and provide accessors if needed. ([squid:ClassVariableVisibilityCheck](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AClassVariableVisibilityCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=39&issues=AWSE1m2gsk71k7lwfRFh&open=AWSE1m2gsk71k7lwfRFh)

[ISSUE_KEY]: AWSE1m2gsk71k7lwfRFh